### PR TITLE
Fix up for merging objects

### PR DESCRIPTION
**Pull request by bryanl**
* Process was using wrong modified document
* Added support for using open api schema
* Merger now can retrieve original object

**note** I'm not 100% sure how to test this, so I did a bunch of manual testing and it didn't seem to break anything.

Signed-off-by: bryanl <bryanliles@gmail.com>

### CODE REVIEW

File: pkg/cluster/tagger.go [hunk 1]
```diff
@@ -16,13 +16,18 @@
...
 	"github.com/ksonnet/ksonnet/pkg/metadata"
 	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 type annotationEncoder interface {
 	EncodePristine(map[string]interface{}) error
+	DecodePristine() (map[string]interface{}, error)
```
Comment (bryanl) [inline pkg/cluster/tagger.go]: Anyone have a better name for this interface?

File: pkg/cluster/merger.go [hunk 2]
```diff
@@ -207,8 +216,11 @@ func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 		return nil, nil, cmdutil.AddSourceToErr(fmt.Sprintf("serializing current configuration from:\n%v\nfor:", obj), source, err)
 	}
 
+	t := newDefaultManaged()
+	original, err := t.Fetch(p.mapping, obj)
+
 	// Retrieve the original configuration of the object from the annotation.
-	original, err := kubectl.GetOriginalConfiguration(p.mapping, obj)
+	// original, err := kubectl.GetOriginalConfiguration(p.mapping, obj)
```
Comment (shomron) [inline pkg/cluster/merger.go]: Can the commented code be removed?

File: pkg/cluster/annotation.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+func (m *defaultAnnotationApplier) Fetch(mapping *meta.RESTMapping, obj runtime.Object) ([]byte, error) {
+	annotations, err := mapping.MetadataAccessor.Annotations(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if annotations == nil {
+		return nil, nil
```
Comment (shomron) [inline pkg/cluster/annotation.go]: Can we document that a nil slice is a valid return value?

File: pkg/cluster/merger.go (hunk 2 above)
Comment (bryanl) [reply pkg/cluster/merger.go]: it should. good catch

**Pull request merged**